kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -171,15 +173,32 @@ private static void create8021xTls(NetworkProperties props, String deviceId, Map
         String identity = props.get(String.class, "net.interface.%s.config.802-1x.identity", deviceId);
         settings.put("identity", new Variant<>(identity));
 
-        String clientCert = props.get(String.class, "net.interface.%s.config.802-1x.client-cert", deviceId);
-        settings.put("client-cert", new Variant<>(clientCert.getBytes(StandardCharsets.UTF_8)));
+        try {
+            Certificate clientCert = props.get(Certificate.class, "net.interface.%s.config.802-1x.client-cert-name",
+                    deviceId);
+            settings.put("client-cert", new Variant<>(clientCert.getEncoded()));
+            logger.info(identity + " client cert: " + clientCert.getEncoded());
+        } catch (Exception e) {
+            logger.error("Enable to find or decode Client Certificate", e);
+        }
+
+        try {
+            PrivateKey privateKey = props.get(PrivateKey.class, "net.interface.%s.config.802-1x.private-key-name",
+                    deviceId);
+            settings.put("private-key", new Variant<>(privateKey.getEncoded()));
+            logger.info(identity + " private key: " + privateKey.getEncoded());
+        } catch (Exception e) {
+            logger.error("Enable to find or decode Private Key", e);
+        }
 
-        String privateKey = props.get(String.class, "net.interface.%s.config.802-1x.private-key", deviceId);
-        settings.put("private-key", new Variant<>(privateKey.getBytes(StandardCharsets.UTF_8)));
+        Optional<Password> privateKeyPassword = props.getOpt(Password.class,
+                "net.interface.%s.config.802-1x.private-key-password", deviceId);
 
-        String privateKeyPassword = props
-                .get(Password.class, "net.interface.%s.config.802-1x.private-key-password", deviceId).toString();
-        settings.put("private-key-password", new Variant<>(privateKeyPassword));
+        if (privateKeyPassword.isPresent()) {
+            settings.put("private-key-password", new Variant<>(privateKeyPassword));
+        }
```

Comment:
Prefer the `ifPresent` syntax

```suggestion
        privateKeyPassword.ifPresent(value -> {
            settings.put("private-key-password", new Variant<>(value));
        });
```